src/main/java/io/confluent/kafkarest/SimpleConsumerManager.java

```diff
@@ -15,108 +15,83 @@
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.Deserializer;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.ws.rs.core.Response;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Properties;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.*;
 
 public class SimpleConsumerManager {
 
   private static final Logger log = LoggerFactory.getLogger(SimpleConsumerManager.class);
 
-  private final int maxPoolSize;
-  private final int poolInstanceAvailabilityTimeoutMs;
-  private final Time time;
-
   private final MetadataObserver mdObserver;
-  private final SimpleConsumerFactory simpleConsumerFactory;
 
-  private final ConcurrentMap<Broker, SimpleConsumerPool> simpleConsumersPools;
+  private final SimpleConsumerPool simpleConsumersPool;
 
-  private AtomicInteger correlationId = new AtomicInteger(0);
-
-  private final Decoder<Object> avroDecoder;
-  private final Decoder<byte[]> binaryDecoder;
-  private final Decoder<Object> jsonDecoder;
+  private final Deserializer<Object> avroDeserializer;
 
   public SimpleConsumerManager(final KafkaRestConfig config,
                                final MetadataObserver mdObserver,
-                               final SimpleConsumerFactory simpleConsumerFactory) {
+                               final ConsumerFactory<byte[], byte[]> simpleConsumerFactory) {
 
     this.mdObserver = mdObserver;
-    this.simpleConsumerFactory = simpleConsumerFactory;
-
-    maxPoolSize = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_MAX_POOL_SIZE_CONFIG);
-    poolInstanceAvailabilityTimeoutMs = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_POOL_TIMEOUT_MS_CONFIG);
-    time = config.getTime();
 
-    simpleConsumersPools = new ConcurrentHashMap<Broker, SimpleConsumerPool>();
+    int maxPoolSize = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_MAX_POOL_SIZE_CONFIG);
+    int poolInstanceAvailabilityTimeoutMs = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_POOL_TIMEOUT_MS_CONFIG);
+    int maxPollTime = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_MAX_POLL_TIME_CONFIG);
+    Time time = config.getTime();
 
-    // Load decoders
-    Properties props = new Properties();
-    props.setProperty("schema.registry.url",
-        config.getString(KafkaRestConfig.SCHEMA_REGISTRY_URL_CONFIG));
-    avroDecoder = new KafkaAvroDecoder(new VerifiableProperties(props));
+    ConsumerFactory<byte[], byte[]> consumerFactory = simpleConsumerFactory;
 
-    binaryDecoder = new DefaultDecoder(new VerifiableProperties());
+    if (consumerFactory == null) {
```

Comment:
It seems that this code path is never used? This constructor is only instantiated from one place and it passes in a non-null `simpleConsumerFactory`